src/deepsparse/transformers/engines/nl_decoder_engine.py

```diff
@@ -335,7 +340,12 @@ def _initialize_kv_cache_state(self, length: int) -> Dict[str, numpy.ndarray]:
         ]
 
         empty_kv_cache_tensor = numpy.zeros(
-            (batch_size, num_attention_heads, length, hidden_dims),
+            (
+                batch_size if not empty else 0,
+                num_attention_heads,
+                length,
+                hidden_dims,
+            ),  # effective array size 0
```

Comment:
Isn't the effective array size only 0 if empty is true? This comment seems a bit misleading if so

Reply:
@dbogunowicz looks like this was just left from the refactor from the WIP